### PR TITLE
Improve project set up

### PR DESCRIPTION
**Pull request by jonathonhawkins92**
## Updated
- `README.md` Running the app locally steps

## Added
- An `.nvmrc` so now when someone starts to use the project, they know what node version everything is built with and can run `nvm i` to download and install it.
- A `.env.example` file outlining the environment variables used in the project with links to what they are used with, this includes a default local DB string.
- New database commands that use the values in `.env` to set up the local database.
  - `db:up` to create the local db
  - `db:down` to destroy the local db, note the data is saved to a volume.
- `README.md` Local database setup steps

### CODE REVIEW

**Comment by jonathonhawkins92** [comment]
@DinerIsmail, I didn't add a NPM script to seed the dev database, and instead opting to documented it, it feels too dangerous to have as an easily accessible command, thoughts?

**Comment by jonathonhawkins92** [comment]
@DinerIsmail could you deploy the feature branch, I would like to see the behaviour you described with the subdomain mismatch, I have an idea to fix it I'd like to test out.

**Comment by DinerIsmail** [comment]
@jonathonhawkins92 Done. Just sent you an invite to the Vercel team as well, hopefully deployments won't require my authorisation once you're a team member.

**Comment by jonathonhawkins92** [comment]
> @DinerIsmail could you deploy the feature branch, I would like to see the behaviour you described with the subdomain mismatch, I have an idea to fix it I'd like to test out.

Well that didn't do what I expected, I assumed they would have something like `${your sub domain}.${random vercel sub domain}.vercel.app` in which case some regex magic would work, rubbish!

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Authentication
+# Package - https://next-auth.js.org/
+NEXTAUTH_URL=
```
Comment (DinerIsmail) [inline .env.example:3]: I think this can be `http://localhost:3000` for running locally

File: package.json [hunk 2]
```diff
@@ -7,6 +7,7 @@
     "build": "next build",
     "start": "next start -H 0.0.0.0 -p ${PORT:-8080}",
     "db": "prisma studio",
+    "db:up": "docker run --name resilience-web -e POSTGRES_PASSWORD=password -p 5432:5432 -d postgres",
```
Comment (DinerIsmail) [inline package.json:10]: Nice, this is really useful!

Comment (jonathonhawkins92) [reply package.json]: Updated to use a more configurable docker-compose file

File: .env.example (hunk 1 above)
Comment (jonathonhawkins92) [reply .env.example]: Done

**Comment by jonathonhawkins92** [comment]
> @jonathonhawkins92 Done. Just sent you an invite to the Vercel team as well, hopefully deployments won't require my authorisation once you're a team member.

Tested and working, thank you 😁

**Comment by DinerIsmail** [comment]
> @DinerIsmail, I didn't add a NPM script to seed the dev database, and instead opting to documented it, it feels too dangerous to have as an easily accessible command, thoughts?

Isn't the seeding already happening by default when Prisma is setup?

**Comment by jonathonhawkins92** [comment]
> > @DinerIsmail, I didn't add a NPM script to seed the dev database, and instead opting to documented it, it feels too dangerous to have as an easily accessible command, thoughts?
> 
> Isn't the seeding already happening by default when Prisma is setup? 

I don't think so, unless I'm missing a trick? What do you think is running? I can see there is a post install script that "generates artifacts" but I'm unsure what that means

**Review by DinerIsmail — APPROVED** [review]
I took this for a spin and it works great! I just added an extra line in the Readme about database seeding with `npx prisma db seed`.

**Pull request merged**